frontend/src/scenes/dashboard/dashboardLogic.tsx

```diff
@@ -929,31 +937,41 @@ export const dashboardLogic = kea<dashboardLogicType>({
                 return
             }
 
-            let breakpointTriggered = false
+            let cancelled = false
             actions.setRefreshStatuses(
                 insights.map((item) => item.short_id),
                 true
             )
 
+            // If a query is in progress, kill that query
+            // we will use one abort controller for all insight queries for this dashboard
+            if (cache.abortController) {
+                cache.abortController.abort()
+            }
+            cache.abortController = new AbortController()
+            const methodOptions: ApiMethodOptions = {
+                signal: cache.abortController.signal,
+            }
+
             const refreshStartTime = performance.now()
-            dashboardQueryId = dashboardQueryId ?? uuid()
+
             let refreshesFinished = 0
             let totalResponseBytes = 0
 
             // array of functions that reload each item
             const fetchItemFunctions = insights.map((insight) => async () => {
-                // :TODO: Support query cancellation and use this queryId in the actual query.
-                const queryId = `${dashboardQueryId}::${uuid()}`
+                const queryId = `dash-${dashboardQueryId}::${uuid()}`
```

Comment:
I don't think the KILL logic below works due to the `dash-` prefix here. The backend checks whether the prefix of the query_id matches and we're not adding the `dash-` when you call cancel.

Suggestion: remove this.

Reply:
> I think the KILL logic below doesn't works due to the dash- prefix here

Ah, dang, yes. I added this, tested it, and then moved what id I was cancelling on 🤦